### PR TITLE
fix(pkg/client/auditor): fix and enhance state signature verification

### PR DESCRIPTION
**Pull request by padurean**
Signed-off-by: Valentin Padurean (Ogg) <912518+padurean@users.noreply.github.com>

### CODE REVIEW

File: pkg/client/auditor/auditor.go [hunk 1]
```diff
@@ -299,6 +296,12 @@ func (a *defaultAuditor) audit() error {
 			return noErr
 		}
 
+		if err := a.verifyStateSignature(serverID, prevState); err != nil {
```
Comment (jeroiraz) [inline pkg/client/auditor/auditor.go:299]: I think this additional signature verification is not needed. Verifying signature before accepting state evolution should be enough.
Verification of local state will double the amount of signature verifications and will make it harder to deal with changing server signing keys. If verification only happens when a new state is validated, it's easier to deal with a new key by just changing the public key at the auditor.

File: pkg/client/auditor/auditor.go [hunk 2]
```diff
@@ -380,6 +383,37 @@ func (a *defaultAuditor) audit() error {
...
+		return fmt.Errorf(
+			"a server signing public key has been specified for the auditor, "+
+				"but the state %s at TX %d received from server %s @ %s is not signed",
+			serverState.GetTxHash(), serverState.GetTxId(), serverID, a.serverAddress)
+	}
+
+	if serverState.GetSignature() != nil {
+		pk := a.serverSigningPubKey
+		if pk == nil {
+			x, y := elliptic.Unmarshal(elliptic.P256(), serverState.GetSignature().GetPublicKey())
```
Comment (byo) [inline pkg/client/auditor/auditor.go:401]: Can we issue some kind of warning here? Checking the signature here without any form of public key verification does not protect from man-in-the-middle attacks and may give a false sense of security.

File: pkg/client/auditor/auditor.go (hunk 1 above)
Comment (padurean) [reply pkg/client/auditor/auditor.go]: Fixed: i removed the signature verification for the previous state.

File: pkg/client/auditor/auditor.go [hunk 3]
```diff
@@ -380,6 +377,37 @@ func (a *defaultAuditor) audit() error {
...
```
Comment (mmeloni) [inline pkg/client/auditor/auditor.go:395]: The code here is binded with elliptic p256 curve type. 
We should use an abstraction.
Inside Signer package there is an ecdsa implentation, we should modify Signer interface and expose UnmarshalKey method and then use it there.

**Review by mmeloni — CHANGES_REQUESTED** [review]

**Review by mmeloni — APPROVED** [review]

File: pkg/client/auditor/auditor.go (hunk 2 above)
Comment (byo) [reply pkg/client/auditor/auditor.go:395]: @padurean We'll be merging the PR as is, please prepare another PR that fixes this and @mmeloni 's comments

**Pull request merged**